packages/nodejs/test/registry.ts

```diff
@@ -0,0 +1,46 @@
+import { Span, SpanOptions, SpanContext, SpanData } from "../src/interfaces"
+import { RootSpan, ChildSpan } from "../src/span"
+
+export class TestRegistry {
+  static spans: any[] = []
+
+  static addSpan(span: any) {
+    this.spans.push(span)
+  }
+
+  static lastSpan(): Span {
+    const length = this.spans.length
+    return this.spans[length - 1]
+  }
+
+  static clear() {
+    this.spans = []
+  }
+}
+
+jest.mock("../src/span", () => {
```

Comment:
I think the fact that requiring this file automatically mocks the `ChildSpan` class for you is somewhat unexpected. Maybe this could be moved to a `.mock` function in the registry that we explicitly call in the test's `.beforeAll()`?

It would also be nice, in turn, to have an `.unmock` function that calls `jest.unmock("../src/span")`, which we can explicitly call from the test's `.afterAll()`.

Reply:
Also, I'm not sure what to make of the note in Jest's documentation that says:

> Modules that are mocked with `jest.mock` are mocked only for the file that calls `jest.mock`. Another file that imports the module will get the original implementation even if it runs after the test file that mocks the module.

Because, uh, this is working, right? So this is clearly not true. I wonder if by "file" they mean "test file"?

Reply:
> {before/after}All, not {before/after}Each -- so, I meant to mock when the describe block containing the tests starts, and unmock when it ends. Jest's naming is confusing.

You mean to unmock at the end of every test file? Jest does that automatically. No need to manually unmock.

> I found a workaround for that! Instead of keeping a list of closed spans and mocking out the closing, keep the state from before the span is closed and use it after it's closed. [Have a look!](https://github.com/appsignal/appsignal-nodejs/compare/http-outgoing-tests...http-outgoing-tests-keep-state-close)

Ah you found a way to make it work! I was trying to implement that, the same way it [works in the Ruby gem tests suite](https://github.com/appsignal/appsignal-ruby/blob/3f503ade83f22f4b0d86d76ea00e5f4dd3c56b6f/spec/support/testing.rb#L79), but I kept running into issues with calling super in the constructor like `const span = super(...)` and adding new fields on the class. I see in my latest version that's no longer an issue and we can apply your change. Can you fix up the commit for a PR based on this one? Or just push it on this branch.

Reply:
Pushed it to this branch.

Reply:
@unflxw looks good. If you're happy with it, please approve the PR so we can merge it.